packages/vite/src/node/plugins/importMetaGlob.ts

```diff
@@ -477,10 +509,17 @@ export async function transformGlobImport(
             originalLineBreakCount > 0
               ? '\n'.repeat(originalLineBreakCount)
               : ''
+          let replacement = ''
+          if (onlyKeys) {
+            replacement = `{${objectProps.join(',')}${lineBreaks}}`
+          } else if (onlyValues) {
+            replacement = `[${objectProps.join(',')}${lineBreaks}]`
+          } else {
```

Comment:
I think we need to keep the `Object.assign` here, it's to fix https://github.com/vitejs/vite/pull/3682. It seems to also work fine with arrays.